src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp

```diff
@@ -169,32 +172,101 @@ void BrgemmKernelExecutor::update_config(const ov::snippets::lowered::Expression
+                        "Incorrect Loop by Brgemm dimension N");
+        M = current_expanded_loop_info->get_increment();
         input_pds[0]->set_subtensor_dim(1, M);
         output_pds[0]->set_subtensor_dim(1, M);
     }
 
-    if (ov::snippets::utils::is_full_dim_value(K)) {
-        K = *in0_shape.rbegin();
-    } else if (ov::snippets::utils::is_dynamic_value(K)) {
-        OPENVINO_THROW("Dynamic K is not supported");
-    }
-
+    /* ------- Dimension N ----------*/
     if (ov::snippets::utils::is_full_dim_value(N)) {
         N = *in1_shape.rbegin();
-    } else if (ov::snippets::utils::is_dynamic_value(N)) {
-        OPENVINO_THROW("Dynamic N is not supported");
+    } else {
+        const auto& current_expanded_loop_info = get_loop_info();
+        const auto& in_ports = current_expanded_loop_info->get_input_ports();
+        const auto& out_ports = current_expanded_loop_info->get_output_ports();
+        // Quick validation check: Should we check that port is really Brgemm port?
+        OPENVINO_ASSERT(in_ports.size() == 2 && !in_ports.front().is_incremented && in_ports.back().is_incremented && in_ports.back().dim_idx == 0 &&
+                        out_ports.size() == 1 && out_ports.front().is_incremented && out_ports.front().dim_idx == 0,
+                        "Incorrect Loop by Brgemm dimension N");
+        N = current_expanded_loop_info->get_increment();
+        input_pds[1]->set_subtensor_dim(0, N);
+        output_pds[0]->set_subtensor_dim(0, N);
+    }
+
+    /* ------- Dimension K ----------*/
+    // 1. If Brgemm block processes full dimension K -> `beta = 0`
+    // 2. If Brgemm block processes part of the dimension K (there is blocking), need to find
+    //    the most first executed Brgemm Block in Loops which iterate through dimension K (work_amount > 0).
+    //    First of them will have `beta = 0`, other - `beta = 1`
+    float beta = 0;
+    if (ov::snippets::utils::is_full_dim_value(K)) {
+        K = *in0_shape.rbegin();
+    } else {
+        const auto& current_expanded_loop_info = get_loop_info();
+        const auto& in_ports = current_expanded_loop_info->get_input_ports();
+        const auto& out_ports = current_expanded_loop_info->get_output_ports();
+        // Quick validation check: Should we check that port is really Brgemm port?
+        OPENVINO_ASSERT(in_ports.size() == 2 && in_ports.front().is_incremented && in_ports.front().dim_idx == 0 &&
+                                                in_ports.back().is_incremented && in_ports.back().dim_idx == 1 &&
+                        out_ports.size() == 1 && !out_ports.front().is_incremented,
+                        "Incorrect Loop by Brgemm dimension K");
+        K = current_expanded_loop_info->get_increment();
+        input_pds[0]->set_subtensor_dim(0, K);
+        input_pds[1]->set_subtensor_dim(1, K);
+
+        // Find all Expanded loops with the same Unified loop information -> they were decomposed from this Unified Loop.
```

Comment:
I like the new approach, thanks for the refactoring 😄 
Do you think is it worth to extract beta computation logic to a separate function? This could make `update_config`'s code a bit more understandable (from my perspective), and this logic will be probably reused for TPP case